os/arch/arm/src/common/up_createstack.c

```diff
@@ -198,14 +198,19 @@ int up_create_stack(FAR struct tcb_s *tcb, size_t stack_size, uint8_t ttype)
 		 * in user threads and null for kernel threads.
 		 */
 		if (!uheap) {
+#ifdef CONFIG_MPU_STACK_OVERFLOW_PROTECTION	/* address must be aligned to 32 byte for MPU region */
+			stack_alloc_size = stack_alloc_size + CONFIG_MPU_STACK_GUARD_SIZE;
+			tcb->stack_alloc_ptr = (uint32_t *)kmm_memalign(32, stack_alloc_size);
+#else
 			tcb->stack_alloc_ptr = (uint32_t *)kmm_malloc(stack_alloc_size);
+#endif
 		} else
 #endif
 		{
 			/* Use the user-space allocator if this is a task or pthread */
 #ifdef CONFIG_MPU_STACK_OVERFLOW_PROTECTION	/* address must be aligned to 32 byte for MPU region */
 			stack_alloc_size = stack_alloc_size + CONFIG_MPU_STACK_GUARD_SIZE;
-			tcb->stack_alloc_ptr = (uint32_t *)kumm_memalign(CONFIG_MPU_STACK_GUARD_SIZE, stack_alloc_size);
+			tcb->stack_alloc_ptr = (uint32_t *)kumm_memalign(32, stack_alloc_size);
```

Comment:
We cannot hardcode the alignment to 32 bytes. In case of v7, the alignment must be same as guard size. In case of v8, the alignment of 32 bytes is sufficient.

Reply:
ok.. I will make this change